src/client/eppo-client.ts

```diff
@@ -151,7 +161,30 @@ export default class EppoClient {
     this.banditModelConfigurationStore = banditModelConfigurationStore;
     this.overrideStore = overrideStore;
     this.configurationRequestParameters = configurationRequestParameters;
-    this.isObfuscated = isObfuscated;
+    this.expectObfuscated = isObfuscated;
+  }
+
+  private maybeWarnAboutObfuscationMismatch(configObfuscated: boolean) {
+    // Don't warn again if we did on the last check.
+    if (configObfuscated !== this.expectObfuscated && !this.obfuscationMismatchWarningIssued) {
+      this.obfuscationMismatchWarningIssued = true;
+      logger.warn(
+        `[Eppo SDK] configuration obfuscation [${configObfuscated}] does not match expected [${this.expectObfuscated}]`,
+      );
+    } else if (configObfuscated === this.expectObfuscated) {
+      // Reset the warning to false in case the client configuration (re-)enters a mismatched state.
+      this.obfuscationMismatchWarningIssued = false;
+    }
+  }
+
+  private isObfuscated() {
+    if (this.configObfuscatedCache === undefined) {
+      this.configObfuscatedCache = OBFUSCATED_FORMATS.includes(
+        this.flagConfigurationStore.getFormat() ?? FormatEnum.SERVER,
+      );
+    }
+    this.maybeWarnAboutObfuscationMismatch(this.configObfuscatedCache);
+    return this.configObfuscatedCache;
```

Comment:
minor: maintaining this cache is probably as costly as doing two comparisons (we actually only need one as precomputed format is not valid here).

major: cache invalidation is also dubious and there seems to be a bug lurking in it. Cache is reset before issues a configuration fetch. However, as fetch is async, the cache will get re-populated if evaluation is requested, and subsequent configuration fetch success does not invalidate the cache. This may cause an evaluation bug if initial configuration obfuscation does not match remote configuration obfuscation (e.g., client initialized with non-obfuscated server config first and fetches an obfuscated configuration later)

If we want to cache obfuscation status, configuration itself is the right place to do that